src/core/utils/index.ts

```diff
@@ -37,4 +37,5 @@ export {
   IContextProvider,
 } from './context';
 export { DEFAULT_APP_CATEGORIES } from './default_app_categories';
-export { WORKSPACE_TYPE } from './constants';
+export { WORKSPACE_PATH_PREFIX, WORKSPACE_TYPE } from './constants';
+export { getWorkspaceIdFromUrl, formatUrlWithWorkspaceId, cleanWorkspaceId } from './workspace';
```

Comment:
Not something we need to decide now, but do you think we should make workspaces a part of core? Right now this PR has a lot of changes in core that are workspace specific. Maybe we just want to keep the UI layer of workspaces in the plugin while the core functionality stays in core. I havent reviewed a lot of the other workspace PR's so this might be what you are already doing so ignore me if thats the case :)

Reply:
Actually we have a workspace module in core (https://github.com/opensearch-project/OpenSearch-Dashboards/blob/main/src/core/public/workspace/workspaces_service.ts), which simply returns some observables.

I guess you want to make these constants and functions into the core module, but unluckily these utils will be used in both public side and server side, so we had to put it under directory `core/utils`.